src/mod/dns.mod/coredns.c

```diff
@@ -1201,6 +1204,103 @@ static void dns_lookup(sockname_t *addr)
   sendrequest(rp, T_PTR);
 }
 
+static int dns_hosts(char *hostn) {
+  #define PATH "/etc/hosts"
+  int fd, len, i, found = 0;
+  struct stat sb;
+  char *addr, hostn_lower[256], hostn_upper[256], *last_newline, *last_hash = NULL, *c, *c2, ip[256];
+  sockname_t name;
+
+  if (!*hostn) {
+    ddebug0(RES_MSG "bogus empty hostname input");
+    return 1;
+  }
+  len = strlen(hostn);
+  if (len > 255) {
+    ddebug0(RES_MSG "bogus len of hostname > 255 input");
+    return 1;
+  }
+  /* due to strncasecmp() and strncmp() being slow if used in loop, precalculate
+   * lower and upper string from hostn and compare with handcrafted code */
+  for (i = 0; i < len; i++) {
+      if (isspace(hostn[i])) {
+        ddebug0(RES_MSG "bogus white-space hostname input");
+        return 1;
+      }
+      hostn_lower[i] = tolower(hostn[i]);
```

Comment:
`tolower((unsigned char)hostn[i])`, same with toupper, as described in man page

Reply:
Fixed tolower() and toupper() usage